syntax = "proto3";

enum Ecommand {
  NoCommand = 0;
  Place = 1;
  Move = 2;
  Capture = 3;
  WhereToMove = 4;
  WhereToCapture = 5;
  Reset = 6;
  Undo = 7;
  ProxyPlayer = 8;
  Time = 9;
  StartGame = 10;
  Strategy = 11;
  Win = 12;
  OpponentDisconnect = 13;
  ServerDisconnect = 14;
  StartLobby = 15;
  GameAlreadyExists = 16;
  Error = 17;
  State=18;
}

message Command {
  Ecommand command = 1;

  oneof payload {
    MovePiece move = 2;
    PlacePiece place = 3;
    CapturePiece capture=4;
    WhereToMovePiece whereToMovePiece=5;
    SetStrategy setStrategy=6;
    GetTime getTime=7;
    GetProxyPlayer getproxyPlayer =8;
    UndoAction undoAction=9;
    GetState getState=10;
    GetWinner getWinner=11;
    GetError getError=12;
  }
}

message GetProxyPlayer{
    int32 player=1;
}

message GetError{
    int32 error=1;
}

message MovePiece {
  int32 from_x = 1;
  int32 from_y = 2;
  int32 to_x = 3;
  int32 to_y = 4;
  int32 player=5;
}

message GetState{
    int32 state=1;
    int32 player=2;

}
message GetWinner{
    int32 state=1;
}

message PlacePiece {
  int32 x = 1;
  int32 y = 2;
  int32 player=3;
}

message CapturePiece {
  int32 x = 1;
  int32 y = 2;
}

message WhereToMovePiece {
  int32 x = 1;
  int32 y = 2;
  repeated IntPair pairs=3;
}

message SetStrategy {
  int32 x = 1;
}

message GetTime{
    int32 timePlayer1=1;
    int32 timePlayer2=2;
    int32 timeAction=3;
}
message IntPair {
  int32 first = 1;
  int32 second = 2;
}

message UndoAction{
  int32 state = 1;
  int32 player = 2;
  repeated IntPair position=3;
}